apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -22,10 +23,19 @@ export class TriggerEvent {
     private jobRepository: JobRepository,
     private addJobUsecase: AddJob,
     private notificationRepository: NotificationRepository,
-    protected createExecutionDetails: CreateExecutionDetails
+    protected createExecutionDetails: CreateExecutionDetails,
+    private logger: PinoLogger
   ) {}
 
   async execute(command: TriggerEventCommand) {
+    const child = this.logger.logger.child({
```

Comment:
What does the creation of the child mean here? 
In addition, what do you think about creating usecase base class in order to assign more info like where the log printed (the usecase name for example)

Reply:
Was just a test we've done, was removed :)